tests/specs/transactions/transactions.spec.ts

```diff
@@ -63,10 +51,14 @@ test.describe('Transaction signing', () => {
 
       if (!displayedFee) throw new Error('Cannot pull fee from UI');
 
-      const [_, requestBody] = await Promise.all([
-        transactionRequestPage.clickConfirmTransactionButton(),
-        interceptTransactionBroadcast(transactionRequestPage.page),
-      ]);
+      await transactionRequestPage.page.route('*/**/v2/transactions', async (route, request) => {
+        requestBody = request.postDataBuffer();
+        await route.fulfill({ status: 200 });
+      });
```

Comment:
This is a bit sketchy imo, it mixes async/sync patterns and relies on a global variable.

Why not modify fn we had before to do both the intercept, mock, and return of requestBody?

```
function mockTransactionBroadcastResponse() {
  return new Promise(async (resolve, reject) => {
    await transactionRequestPage.page.route(
      "*/**/v2/transactions",
      async (route, request) => {
        await route.fulfill({ status: 200 });
        resolve(request.postDataBuffer());
      }
    );
  })
}

```

Reply:
or maybe you can just return it, if the `page.route` also passes it back

Reply:
Yeah, let me try this again. I think I did start there but was struggling to get it to work. I tried a number of things learning how to intercept it correctly, but this might work now that I was successful with the current solution. I'll try this today to see if works.

Reply:
@kyranjamie I can't get your suggestion to work. The test fails. How is what I've done different from how we use `let` variables to assign pages in the async beforeEach?

Reply:
@kyranjamie nvm here, I figured out the right pattern after doing some digging:

```ts
const requestPromise = transactionRequestPage.page.waitForRequest('*/**/v2/transactions');
const request = await requestPromise;
const requestBody = request.postDataBuffer();
```

Reply:
That allows us to use the request in tests, but catch the route and fulfill it w/out broadcasting the tx.